packages/editor/src/lib/utils/shapeNavigation.ts

```diff
@@ -0,0 +1,199 @@
+import { TLShape, TLShapeId } from '@tldraw/tlschema'
+import { Editor } from '../editor/Editor'
+import { Vec } from '../primitives/Vec'
+
+const SHALLOW_ANGLE = 20
+const ROW_THRESHOLD = 100
+
+/** @public */
+export type AdjacentDirection = 'next' | 'prev' | 'left' | 'right' | 'up' | 'down'
+
+/** @public */
+export function selectAdjacentShape(editor: Editor, direction: AdjacentDirection) {
```

Comment:
Instead of having these as free functions, could we either make these methods on `Editor` or put them in a `manager` like `editor.snaps`? So you'd have either `editor.selectAdjacentShape(...)` or `editor.navigation.selectAdjacentShape()`? I think free functions are probably the least discoverable of the possible API options here, they're no good for autocomplete-driven-development

Reply:
yeah, I don't disagree! i'd prefer it honestly it on the editor as well. (even better for `editor.navigation.selectAdjacentShape()` but it feels like we always do a flat structure, with maybe camera as the exception?)

sometimes i can't tell if we just don't like putting too many things on the editor